src/ui/common/src/components/workflows/workflowStatus.tsx

```diff
@@ -54,30 +23,105 @@ export const StatusIndicator: React.FC<Props> = ({ status }) => {
       backgroundColor = theme.palette.Info;
       break;
     case ExecutionStatus.Registered:
-      // TODO: Figure out color to use for Registered.
-      backgroundColor = theme.palette.Info;
+      backgroundColor = theme.palette.Registered;
       break;
     case ExecutionStatus.Running:
-      // TODO: Figure out color to use for running
-      backgroundColor = theme.palette.Info;
+      backgroundColor = theme.palette.Running;
       break;
     case ExecutionStatus.Succeeded:
       backgroundColor = theme.palette.Success;
       break;
     case ExecutionStatus.Unknown:
+    default:
       backgroundColor = theme.palette.gray[400];
       break;
   }
 
+  return backgroundColor;
+};
+
+export const getExecutionStatusLabel = (status: ExecutionStatus): string => {
+  let labelText = 'Succeeded';
+  switch (status) {
+    case ExecutionStatus.Canceled:
+      labelText = 'Canceled';
+      break;
+    case ExecutionStatus.Failed:
+      labelText = 'Failed';
+      break;
+    case ExecutionStatus.Pending:
+      labelText = 'Pending';
+      break;
+    case ExecutionStatus.Registered:
+      labelText = 'Registered';
+      break;
+    case ExecutionStatus.Running:
+      labelText = 'Running';
+      break;
+    case ExecutionStatus.Succeeded:
+      labelText = 'Succeeded';
+      break;
+    case ExecutionStatus.Unknown:
+      labelText = 'Unknown';
+      break;
+    default:
+      labelText = 'Unknown';
+      break;
+  }
+
+  return labelText;
+};
+
+export const StatusChip: React.FC<Props> = ({ status }) => {
+  const statusIcons = [];
+
+  const getStatusChipTextColor = (status: ExecutionStatus): string => {
+    let textColor = theme.palette.black;
+    switch (status) {
+      case ExecutionStatus.Canceled:
+      case ExecutionStatus.Unknown:
+      case ExecutionStatus.Running:
+        textColor = theme.palette.black;
+        break;
+      default:
+        textColor = theme.palette.white;
+        break;
+    }
+
+    return textColor;
+  };
+
+  statusIcons.push();
+
   return (
-    <Box
+    <Chip
+      label={getExecutionStatusLabel(status)}
       sx={{
-        width: '8px',
-        height: '8px',
-        backgroundColor,
-        borderRadius: 999,
-        alignSelf: 'center',
+        backgroundColor: getExecutionStatusColor(status),
+        color: getStatusChipTextColor(status),
       }}
+      size="small"
     />
   );
 };
+
+export default StatusChip;
+
+/**
```

Comment:
This may seem odd, but this format of comment allows the text to appear as a bullet point under the name of the story in Storybook.

Screenshot:
<img width="1511" alt="image" src="https://user-images.githubusercontent.com/1031759/204668683-a3c56ceb-2627-438e-a558-f7d040729d69.png">